test/test_response.py

```diff
@@ -377,6 +377,59 @@ def test_empty_stream(self):
 
         self.assertRaises(StopIteration, next, stream)
 
+    def test_length_no_header(self):
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, preload_content=False)
+        assert resp.length is None
+
+    def test_length_w_valid_header(self):
+        headers = {"content-length": "5"}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        assert resp.length == 5
+
+    def test_length_w_bad_header(self):
+        garbage = {'content-length': 'foo'}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+        garbage['content-length'] = "-10"
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+    def test_length_when_chunked(self):
+        headers = {'content-length': '5',
+                   'transfer-encoding': 'chunked'}
```

Comment:
This combination of headers is forbidden by [RFC 7230 Section 3.3.2](https://tools.ietf.org/html/rfc7230#section-3.3.2):

> A sender MUST NOT send a Content-Length header field in any message that contains a Transfer-Encoding header field.